src/ray/raylet/scheduling_queue.h

```diff
@@ -37,6 +37,118 @@ enum class TaskState {
+ /// A list of tasks.
+ std::list<Task> task_list_;
+ /// A hash to speed up looking up a task.
+ std::unordered_map<TaskID, std::list<Task>::iterator> task_map_;
+};
+
+class ReadyQueue : public TaskQueue {
+public:
+  /// \brief Append a task to the ready queue.
+  ///
+  /// \param task_id The task ID for the task to append.
+  /// \param task The task to append to the queue.
+  /// \return Whether the append operation succeeds.
+  bool AppendTask(const TaskID &task_id, const Task &task);
+
+  /// \brief Remove a task from the ready queue.
+  ///
+  /// \param task_id The task ID for the task to remove from the queue.
+  /// \param removed_tasks If the task specified by task_id is successfully
+  ///  removed from the queue, the task data is appended to the vector. Can
+  ///  be a nullptr, in which case nothing is appended.
+  /// \return Whether the removal succeeds.
+  bool RemoveTask(const TaskID &task_id,
+                  std::vector<Task> *removed_tasks = nullptr);
+
+  /// \brief Check whether any of the tasks in the ready queue can be scheduled.
+  ///
+  /// \param local_available_resources: resources available on local node.
+  /// \return True if there is at least a task that can be scheduled, i.e.
+  /// whose requirements can be satisfied by the local node; false otherwise.
+  const bool CanScheduleMinTask(const ResourceIdSet &local_available_resources) const;
+
+  const ResourceSet &GetCurrentResourceLoad() const { return current_resource_load_; }
+
+  /// \brief Get resources of the task with the minimum resource requirements
+  /// in the ready queue.
+  ///
+  /// \return: Minimum resources required by any task in the ready queue.
+  const ResourceSet &GetMinTaskResources() const { return min_task_resources_;};
+
+  /// \brief Get number of tasks with minimum resource requirements in the
+  /// ready queue.
+  ///
+  /// \return Number of tasks with minimum resource requirements.
+  const int &GetMinTaskCount() const { return min_task_count_;};
+
+  /// \brief Update the min  resources required by a task in the ready queue,
+  ///  when a new task is added to the queue.
+  ///
+  /// \param resources: Resource requirements of the new task being added.
+  void UpdateMinOnAdd(const ResourceSet &resources);
+
+  /// \brief Update the min resources required by a task in the ready queue,
+  ///  when a task is removed from the queue.
+  ///
+  /// \param resources: Resource requirements of the new task being removed.
+  void UpdateMinOnRemove(const ResourceSet &resources);
+
+  /// \brief Recompute the resources of the tasks with minimum requirements
+  /// in the ready queue.
+  void RecomputeMinResources();
+
+private:
+  /// Aggregate resources of all the tasks in this queue.
+  ResourceSet current_resource_load_;
+  /// Minimum resources requested by any task in tyhe ready queue.
+  ResourceSet min_task_resources_;
+  /// Count of tasks which the minimum resource requirements in the ready queue.
+  int min_task_count_ = 0;
```

Comment:
We should initialize these in `ReadyQueue::ReadyQueue()` ctor

Reply:
Hm, not clear that that buys anything here. The ResourceSet constructor will already be automatically called and the min_task_count_ is initialized.